From d9db1cadcdacbc589a013d6d307c29e27dc07670 Mon Sep 17 00:00:00 2001
From: Chen Liangjun <b36089@freescale.com>
Date: Thu, 23 Aug 2012 12:30:57 +0800
Subject: ENGR00221177 ESAI ASRC: add mutex protection between ESAI and ASRC
 P2P

ESAI playback and ASRC P2P playback use difference codec_dai while using
the same codec. Thus they can't work together.

In this patch, add mutual protection between ESAI playback and ASRC P2P
playback.

Signed-off-by: Chen Liangjun <b36089@freescale.com>
---
 sound/soc/imx/imx-esai.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sound/soc/imx/imx-esai.c b/sound/soc/imx/imx-esai.c
index bc56d80..359cb2a 100644
--- a/sound/soc/imx/imx-esai.c
+++ b/sound/soc/imx/imx-esai.c
@@ -278,6 +278,12 @@ static int imx_esai_startup(struct snd_pcm_substream *substream,
 {
 	struct imx_esai *esai = snd_soc_dai_get_drvdata(cpu_dai);
 
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK &&
+			(local_esai->imx_esai_txrx_state & IMX_DAI_ESAI_TX)) {
+		pr_err("error: too much esai playback!\n");
+		return -EINVAL;
+	}
+
 	if (!(local_esai->imx_esai_txrx_state & IMX_DAI_ESAI_TXRX)) {
 		clk_enable(esai->clk);
 
-- 
1.7.9.5

